### PR TITLE
Remove unused FieldData package and DocValuesFormat class

### PR DESCRIPTION
**Pull request by romseygeek**
The only classes being used in the FieldData package are NullValueOrder, which
can be moved to be part of the sorting-related code, and SortedBinaryDocValues.
This last class is just a thin wrapper around SortedSetDocValues, which has the
unfortunate side-effect of hiding some fairly inefficient ordinal resolving behaviour,
so it can be trivially replaced, and all the other code in the package removed.

Similarly, the only code in DocValuesFormat that is being used is a static helper
method that converts a byte-encoded IP address to a String.  This can be moved
to NetworkUtils, and the rest of DocValuesFormat removed.

### CODE REVIEW

File: extensions/functions/src/main/java/io/crate/operation/aggregation/HyperLogLogDistinctAggregation.java [hunk 1]
```diff
@@ -245,7 +244,10 @@ public DocValueAggregator<?> getDocValueAggregator(LuceneReferenceResolver refer
                     @Override
                     public void apply(RamAccounting ramAccounting, int doc, HllState state) throws IOException {
                         if (super.values.advanceExact(doc) && super.values.docValueCount() == 1) {
-                            BytesRef ref = super.values.nextValue();
+                            // TODO: for small cardinality fields it might be worth hashing all terms
```
Comment (mfussenegger) [inline extensions/functions/src/main/java/io/crate/operation/aggregation/HyperLogLogDistinctAggregation.java:247]: Can you remove these TODOs and instead create an issue?

**Review by mfussenegger — APPROVED** [review]
Nice

File: extensions/functions/src/main/java/io/crate/operation/aggregation/HyperLogLogDistinctAggregation.java (hunk 1 above)
Comment (romseygeek) [reply extensions/functions/src/main/java/io/crate/operation/aggregation/HyperLogLogDistinctAggregation.java:247]: I opened https://github.com/crate/crate/issues/16439

**Review by matriv — COMMENTED** [review]
:+1:

**Pull request merged**